// vue-socials/dist/esm/components/SMastodon/SMastodon.js
import { defineComponent } from 'vue';
import BaseSocials from '../../mixins/BaseSocial/BaseSocial.js';
import getSerialisedParams from '../../utils/getSerialisedParams.js';

/**
 * Hey!
 *
 * SMastodon component used for mastodon social network
 * @link https://mastodon.social/
 */
/**
 * Share parameters for link
 * @link https://aly-ve.github.io/Mastodon-share-button/
 */

var SMastodon = /* #__PURE__ */defineComponent({
  name: 'SMastodon',
  mixins: [BaseSocials('Mastodon', {
    width: 600,
    height: 700
  })],
  computed: {
    networkURL: function networkURL() {
      var shareOptions = this.shareOptions;
      var domain = shareOptions.domain,
          text = shareOptions.text,
          url = shareOptions.url;
      var BASE_URL = "".concat(domain, "/share");
      var serialisedParams = getSerialisedParams({
        text: text,
        url: url
      });
      return "".concat(BASE_URL).concat(serialisedParams);
    }
  },
  render: function render() {
    return this.generateComponent(this.networkURL);
  }
});

export default SMastodon;
