// vue-socials/dist/esm/components/SMicrosoftTeams/SMicrosoftTeams.js
import { defineComponent } from 'vue';
import BaseSocials from '../../mixins/BaseSocial/BaseSocial.js';
import getSerialisedParams from '../../utils/getSerialisedParams.js';

/**
 * Hey!
 *
 * SMicrosoftTeams component used for MicrosoftTeams social network
 * @link https://www.microsoft.com/en/microsoft-365/microsoft-teams/group-chat-software
 */
/**
 * Share parameters for link
 * @link https://docs.microsoft.com/en-us/microsoftteams/platform/concepts/build-and-test/share-to-teams
 */

var SMicrosoftTeams = /* #__PURE__ */defineComponent({
  name: 'SMicrosoftTeams',
  mixins: [BaseSocials('MicrosoftTeams')],
  computed: {
    networkURL: function networkURL() {
      var BASE_URL = 'https://teams.microsoft.com/share';
      var shareOptions = this.shareOptions;
      var url = shareOptions.url,
          preview = shareOptions.preview,
          text = shareOptions.text,
          instruction = shareOptions.instruction,
          title = shareOptions.title;
      var serialisedParams = getSerialisedParams({
        href: url,
        preview: preview,
        assignTitle: title,
        msgText: text,
        assignInstr: instruction
      });
      return "".concat(BASE_URL).concat(serialisedParams);
    }
  },
  render: function render() {
    return this.generateComponent(this.networkURL);
  }
});

export default SMicrosoftTeams;
