// vue-socials/dist/esm/components/SPinterest/SPinterest.js
import { defineComponent } from 'vue';
import BaseSocials from '../../mixins/BaseSocial/BaseSocial.js';
import getSerialisedParams from '../../utils/getSerialisedParams.js';

/**
 * Hey!
 *
 * SPinterest component used for Pinterest social network
 * @link https://www.pinterest.com/
 */
/**
 * Share parameters for link
 * @link https://developers.pinterest.com/docs/widgets/save/?
 */

var SPinterest = /* #__PURE__ */defineComponent({
  name: 'SPinterest',
  mixins: [BaseSocials('Pinterest', {
    width: 700,
    height: 700
  })],
  computed: {
    networkURL: function networkURL() {
      var BASE_URL = 'http://pinterest.com/pin/create/button/';
      var shareOptions = this.shareOptions;
      var url = shareOptions.url,
          media = shareOptions.media,
          description = shareOptions.description;
      var serialisedParams = getSerialisedParams({
        url: url,
        media: media,
        description: description
      });
      return "".concat(BASE_URL).concat(serialisedParams);
    }
  },
  render: function render() {
    return this.generateComponent(this.networkURL);
  }
});

export default SPinterest;
