// vue-socials/dist/esm/components/SEvernote/SEvernote.js
import { defineComponent } from 'vue';
import BaseSocials from '../../mixins/BaseSocial/BaseSocial.js';
import getSerialisedParams from '../../utils/getSerialisedParams.js';

/**
 * Hey!
 *
 * SEvernote component used for Evernote social network
 * @link https://www.evernote.com/
 * @example https://www.evernote.com/clip.action?url=https%3A%2F%2Fgithub.com%2F&title=Title
 */
/**
 * Share parameters for link
 */

var SEvernote = /* #__PURE__ */defineComponent({
  name: 'SEvernote',
  mixins: [BaseSocials('Evernote', {
    width: 800,
    height: 560
  })],
  computed: {
    networkURL: function networkURL() {
      var BASE_URL = 'https://www.evernote.com/clip.action';
      var shareOptions = this.shareOptions;
      var url = shareOptions.url,
          title = shareOptions.title;
      var serialisedParams = getSerialisedParams({
        url: url,
        title: title
      });
      return "".concat(BASE_URL).concat(serialisedParams);
    }
  },
  render: function render() {
    return this.generateComponent(this.networkURL);
  }
});

export default SEvernote;
