// vue-socials/dist/esm/components/SSms/SSms.js
import { defineComponent, h } from 'vue';
import getSerialisedParams from '../../utils/getSerialisedParams.js';

/**
 * Hey!
 *
 * SSms component used for sending SMS via an HTML link
 */
/**
 * Share parameters for link
 */

var SSms = /* #__PURE__ */defineComponent({
  name: 'SSms',
  props: {
    /**
     * Share parameters for SMS
     */
    shareOptions: {
      type: Object,
      required: true
    }
  },
  computed: {
    networkURL: function networkURL() {
      var BASE_URL = 'sms:';
      var shareOptions = this.shareOptions;
      var number = shareOptions.number,
          body = shareOptions.body;
      var serialisedParams = getSerialisedParams({
        body: body
      });
      return "".concat(BASE_URL).concat(number).concat(serialisedParams);
    }
  },
  render: function render() {
    var _this$$slots$default, _this$$slots;

    return h('a', {
      attrs: {
        href: this.networkURL
      }
    }, (_this$$slots$default = (_this$$slots = this.$slots).default) === null || _this$$slots$default === void 0 ? void 0 : _this$$slots$default.call(_this$$slots));
  }
});

export default SSms;
