// vue-socials/dist/esm/components/SOdnoklassniki/SOdnoklassniki.js
import { defineComponent } from 'vue';
import BaseSocials from '../../mixins/BaseSocial/BaseSocial.js';
import getSerialisedParams from '../../utils/getSerialisedParams.js';

/**
 * Hey!
 *
 * SOdnoklassniki component used for Odnoklassniki (English: "Classmates") social network
 * @link https://ok.ru/
 */
/**
 * Share parameters for link
 * @link https://apiok.ru/en/ext/like
 */

var SOdnoklassniki = /* #__PURE__ */defineComponent({
  name: 'SOdnoklassniki',
  mixins: [BaseSocials('Odnoklassniki')],
  computed: {
    networkURL: function networkURL() {
      var BASE_URL = 'https://connect.ok.ru/offer';
      var shareOptions = this.shareOptions;
      var url = shareOptions.url,
          title = shareOptions.title,
          imageUrl = shareOptions.imageUrl;
      var serialisedParams = getSerialisedParams({
        url: url,
        title: title,
        imageUrl: imageUrl
      });
      return "".concat(BASE_URL).concat(serialisedParams);
    }
  },
  render: function render() {
    return this.generateComponent(this.networkURL);
  }
});

export default SOdnoklassniki;
