// vue-socials/dist/esm/components/SReddit/SReddit.js
import { defineComponent } from 'vue';
import BaseSocials from '../../mixins/BaseSocial/BaseSocial.js';
import getSerialisedParams from '../../utils/getSerialisedParams.js';

/**
 * Hey!
 *
 * SReddit component used for Reddit social network
 * @link https://www.reddit.com/
 */
/**
 * Share parameters for link
 */

var SReddit = /* #__PURE__ */defineComponent({
  name: 'SReddit',
  mixins: [BaseSocials('Reddit', undefined, undefined, undefined, false)],
  computed: {
    networkURL: function networkURL() {
      var BASE_URL = 'https://reddit.com/submit';
      var shareOptions = this.shareOptions;
      var url = shareOptions.url,
          title = shareOptions.title,
          text = shareOptions.text,
          selftext = shareOptions.selftext;
      var serialisedParams = getSerialisedParams({
        url: url,
        title: title,
        text: text,
        selftext: selftext
      });
      return "".concat(BASE_URL).concat(serialisedParams);
    }
  },
  render: function render() {
    return this.generateComponent(this.networkURL);
  }
});

export default SReddit;
