// vue-socials/dist/esm/components/SMailRu/SMailRu.js
import { defineComponent } from 'vue';
import BaseSocials from '../../mixins/BaseSocial/BaseSocial.js';
import getSerialisedParams from '../../utils/getSerialisedParams.js';

/**
 * Hey!
 *
 * SMailRu component used for Mail.ru social network
 * @link https://my.mail.ru/
 */
/**
 * Share parameters for link
 * @link https://api.mail.ru/sites/plugins/share/extended/
 */

var SMailRu = /* #__PURE__ */defineComponent({
  name: 'SMailRu',
  mixins: [BaseSocials('Mail.ru')],
  computed: {
    networkURL: function networkURL() {
      var BASE_URL = 'https://connect.mail.ru/share';
      var shareOptions = this.shareOptions;
      var url = shareOptions.url,
          title = shareOptions.title,
          description = shareOptions.description,
          image = shareOptions.image,
          swfUrl = shareOptions.swfUrl,
          width = shareOptions.width,
          height = shareOptions.height;
      var serialisedParams = getSerialisedParams({
        share_url: url,
        title: title,
        description: description,
        image_url: image,
        swfurl: swfUrl,
        width: width,
        height: height
      });
      return "".concat(BASE_URL).concat(serialisedParams);
    }
  },
  render: function render() {
    return this.generateComponent(this.networkURL);
  }
});

export default SMailRu;
