// vue-socials/dist/esm/components/SVkontakte/SVkontakte.js
import { defineComponent } from 'vue';
import BaseSocials from '../../mixins/BaseSocial/BaseSocial.js';
import getSerialisedParams from '../../utils/getSerialisedParams.js';

/**
 * Hey!
 *
 * SVkontakte component used for Vkontakte social network
 * @link https://vk.com
 */
/**
 * Share parameters for link
 * @link https://vk.com/dev/widget_share
 */

var SVkontakte = /* #__PURE__ */defineComponent({
  name: 'SVkontakte',
  mixins: [BaseSocials('Vkontakte', undefined, undefined, undefined, false)],
  computed: {
    networkURL: function networkURL() {
      var BASE_URL = 'https://vk.com/share.php';
      var shareOptions = this.shareOptions;
      var url = shareOptions.url,
          title = shareOptions.title,
          image = shareOptions.image,
          noParse = shareOptions.noParse,
          noVkLinks = shareOptions.noVkLinks;
      var serialisedParams = getSerialisedParams({
        url: url,
        title: title,
        image: image,
        noparse: noParse,
        no_vk_links: Number(noVkLinks)
      });
      return "".concat(BASE_URL).concat(serialisedParams);
    }
  },
  render: function render() {
    return this.generateComponent(this.networkURL);
  }
});

export default SVkontakte;
