// vue-socials/dist/esm/components/SPinterestCount/SPinterestCount.js
import { defineComponent } from 'vue';
import JSONP from '../../utils/jsonp.js';
import getSerialisedParams from '../../utils/getSerialisedParams.js';
import BaseCount from '../../mixins/BaseCount/BaseCount.js';

/**
 * Hey!
 *
 * SPinterestCount component used for Pinterest social network
 * @link https://www.pinterest.com/
 */
/**
 * Share parameters for link
 */

var SPinterestCount = /* #__PURE__ */defineComponent({
  name: 'SPinterestCount',
  mixins: [BaseCount('Pinterest')],
  methods: {
    handlePinterestResponse: function handlePinterestResponse(data) {
      this.handleResult(data);
      this.handleCount(data.count);
    }
  },
  mounted: function mounted() {
    var _this = this;

    var shareOptions = this.shareOptions;
    var url = shareOptions.url;
    var BASE_URL = 'https://api.pinterest.com/v1/urls/count.json';
    var finalURL = "".concat(BASE_URL).concat(getSerialisedParams({
      url: url
    }));
    this.handleLoading(true);
    JSONP(finalURL, function (err, data) {
      _this.handleLoading(false);

      if (err) {
        _this.handleError(err);
      }

      if (data) {
        _this.handlePinterestResponse(data);
      }
    });
  }
});

export default SPinterestCount;
