// vue-socials/dist/esm/components/SRenRen/SRenRen.js
import { defineComponent } from 'vue';
import BaseSocials from '../../mixins/BaseSocial/BaseSocial.js';
import getSerialisedParams from '../../utils/getSerialisedParams.js';

/**
 * Hey!
 *
 * SRenRen component used for RenRen social network
 * @link https://renren.com
 */
/**
 * Share parameters for link
 * @link http://dev.renren.com/website/?widget=rrshare&content=use
 */

var SRenRen = /* #__PURE__ */defineComponent({
  name: 'SRenRen',
  mixins: [BaseSocials('RenRen')],
  computed: {
    networkURL: function networkURL() {
      var BASE_URL = 'http://widget.renren.com/dialog/share';
      var shareOptions = this.shareOptions;
      var resourceUrl = shareOptions.resourceUrl,
          srcUrl = shareOptions.srcUrl,
          title = shareOptions.title,
          description = shareOptions.description,
          pic = shareOptions.pic;
      var serialisedParams = getSerialisedParams({
        resourceUrl: resourceUrl,
        srcUrl: srcUrl,
        title: title,
        description: description,
        pic: pic
      });
      return "".concat(BASE_URL).concat(serialisedParams);
    }
  },
  render: function render() {
    return this.generateComponent(this.networkURL);
  }
});

export default SRenRen;
