// vue-socials/dist/esm/components/SFlipBoard/SFlipBoard.js
import { defineComponent } from 'vue';
import BaseSocials from '../../mixins/BaseSocial/BaseSocial.js';
import getSerialisedParams from '../../utils/getSerialisedParams.js';

/**
 * Hey!
 *
 * SFlipBoard component used for FlipBoard social network
 * @link https://flipboard.com/
 * @example https://share.flipboard.com/bookmarklet/popout?url=https%3A%2F%2Fgithub.com%2F&title=Title
 */
/**
 * Share parameters for link
 * @link https://about.flipboard.com/social-plugins/?tool=browser
 */

var SFlipBoard = /* #__PURE__ */defineComponent({
  name: 'SFlipBoard',
  mixins: [BaseSocials('Flipboard', {
    width: 834,
    height: 572
  })],
  computed: {
    networkURL: function networkURL() {
      var BASE_URL = 'https://share.flipboard.com/bookmarklet/popout';
      var shareOptions = this.shareOptions;
      var url = shareOptions.url,
          title = shareOptions.title;
      var serialisedParams = getSerialisedParams({
        v: '2',
        url: url,
        title: title
      });
      return "".concat(BASE_URL).concat(serialisedParams);
    }
  },
  render: function render() {
    return this.generateComponent(this.networkURL);
  }
});

export default SFlipBoard;
