// vue-socials/dist/esm/components/SInstaPaper/SInstaPaper.js
import { defineComponent } from 'vue';
import BaseSocials from '../../mixins/BaseSocial/BaseSocial.js';
import getSerialisedParams from '../../utils/getSerialisedParams.js';

/**
 * Hey!
 *
 * SInstaPaper component used for InstaPaper  social network
 * @link http://www.instapaper.com/
 */
/**
 * Share parameters for link
 */

var SInstaPaper = /* #__PURE__ */defineComponent({
  name: 'SInstaPaper',
  mixins: [BaseSocials('InstaPaper', {
    width: 600,
    height: 600
  })],
  computed: {
    networkURL: function networkURL() {
      var BASE_URL = 'http://www.instapaper.com/edit';
      var shareOptions = this.shareOptions;
      var url = shareOptions.url,
          title = shareOptions.title,
          description = shareOptions.description;
      var serialisedParams = getSerialisedParams({
        url: url,
        title: title,
        description: description
      });
      return "".concat(BASE_URL).concat(serialisedParams);
    }
  },
  render: function render() {
    return this.generateComponent(this.networkURL);
  }
});

export default SInstaPaper;
