// vue-socials/dist/esm/components/SLine/SLine.js
import { defineComponent } from 'vue';
import BaseSocials from '../../mixins/BaseSocial/BaseSocial.js';
import getSerialisedParams from '../../utils/getSerialisedParams.js';

/**
 * Hey!
 *
 * SLine component used for Line social network
 * @link https://line.me/
 */
/**
 * Share parameters for link
 * @link https://social-plugins.line.me/en/how_to_install#lineitbutton
 */

var SLine = /* #__PURE__ */defineComponent({
  name: 'SLine',
  mixins: [BaseSocials('Line', {
    width: 600,
    height: 600
  })],
  computed: {
    networkURL: function networkURL() {
      var BASE_URL = 'https://lineit.line.me/share/ui';
      var shareOptions = this.shareOptions;
      var url = shareOptions.url,
          text = shareOptions.text;
      var serialisedParams = getSerialisedParams({
        url: url,
        text: text
      });
      return "".concat(BASE_URL).concat(serialisedParams);
    }
  },
  render: function render() {
    return this.generateComponent(this.networkURL);
  }
});

export default SLine;
