// vue-socials/dist/esm/components/STrello/STrello.js
import { defineComponent } from 'vue';
import BaseSocials from '../../mixins/BaseSocial/BaseSocial.js';
import getSerialisedParams from '../../utils/getSerialisedParams.js';

/**
 * Hey!
 *
 * STrello component used for Trello social network
 * @link https://trello.com/
 */
/**
 * Share parameters for link
 */

var STrello = /* #__PURE__ */defineComponent({
  name: 'STrello',
  mixins: [BaseSocials('Trello')],
  computed: {
    networkURL: function networkURL() {
      var BASE_URL = 'https://trello.com/add-card';
      var shareOptions = this.shareOptions;
      var url = shareOptions.url,
          name = shareOptions.name,
          description = shareOptions.description;
      var serialisedParams = getSerialisedParams({
        url: url,
        name: name,
        desc: description,
        mode: 'popup'
      });
      return "".concat(BASE_URL).concat(serialisedParams);
    }
  },
  render: function render() {
    return this.generateComponent(this.networkURL);
  }
});

export default STrello;
