// vue-socials/dist/esm/components/SYammer/SYammer.js
import { defineComponent } from 'vue';
import BaseSocials from '../../mixins/BaseSocial/BaseSocial.js';
import getSerialisedParams from '../../utils/getSerialisedParams.js';

/**
 * Hey!
 *
 * SYammer component used for Yammer social network
 * @link https://yammer.com/
 */
/**
 * Share parameters for link
 * @link https://developer.yammer.com/docs/share-button
 */

var SYammer = /* #__PURE__ */defineComponent({
  name: 'SYammer',
  mixins: [BaseSocials('Yammer', {
    width: 900,
    height: 600
  })],
  computed: {
    networkURL: function networkURL() {
      var BASE_URL = 'https://www.yammer.com/messages/new';
      var shareOptions = this.shareOptions;
      var text = shareOptions.text,
          url = shareOptions.url,
          groupId = shareOptions.groupId;
      var serialisedParams = getSerialisedParams({
        login: true,
        status: "".concat(text, " ").concat(url),
        group_id: groupId
      });
      return "".concat(BASE_URL).concat(serialisedParams);
    }
  },
  render: function render() {
    return this.generateComponent(this.networkURL);
  }
});

export default SYammer;
