// vue-socials/dist/esm/components/SQZone/SQZone.js
import { defineComponent } from 'vue';
import BaseSocials from '../../mixins/BaseSocial/BaseSocial.js';
import getSerialisedParams from '../../utils/getSerialisedParams.js';

/**
 * Hey!
 *
 * SQZone component used for QZone social network
 * @link https://qzone.qq.com/
 */
/**
 * Share parameters for link
 */

var SQZone = /* #__PURE__ */defineComponent({
  name: 'SQZone',
  mixins: [BaseSocials('QZone')],
  computed: {
    networkURL: function networkURL() {
      var BASE_URL = 'http://sns.qzone.qq.com/cgi-bin/qzshare/cgi_qzshare_onekey';
      var shareOptions = this.shareOptions;
      var url = shareOptions.url,
          title = shareOptions.title,
          summary = shareOptions.summary;
      var serialisedParams = getSerialisedParams({
        url: url,
        title: title,
        summary: summary
      });
      return "".concat(BASE_URL).concat(serialisedParams);
    }
  },
  render: function render() {
    return this.generateComponent(this.networkURL);
  }
});

export default SQZone;
